### PR TITLE
[Github] linked branch actions

### PR DESCRIPTION
**Pull request by khasbilegt**
## Description

This change adds create branch command utilizing form view and related actions in created issue as well.

## Screencast

github-5
github-6
github 2023-01-06 at 23 26 34
github 2023-01-06 at 23 37 11

## Checklist

- [x] I read the [extension guidelines](https://developers.raycast.com/basics/prepare-an-extension-for-store)
- [x] I read the [documentation about publishing](https://developers.raycast.com/basics/publish-an-extension)
- [x] I ran `npm run build` and [tested this distribution build in Raycast](https://developers.raycast.com/basics/prepare-an-extension-for-store#metadata-and-configuration)
- [x] I checked that files in the `assets` folder are used by the extension itself
- [x] I checked that assets used by the `README` are placed outside of the `metadata` folder

### CODE REVIEW

File: extensions/github/src/components/IssueActions.tsx [hunk 1]
```diff
@@ -182,6 +246,15 @@ export default function IssueActions({ issue, mutateList, mutateDetail, children
 
         <AddProjectSubmenu issue={issue} mutate={mutate} />
 
+        {!linkedBranch ? (
+          <Action
+            title={"Create a branch"}
```
Comment (pernielsentikaer) [inline extensions/github/src/components/IssueActions.tsx:251]:
```suggestion
            title={"Create a Branch"}
```

File: extensions/github/src/create-branch.tsx [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+
+  useEffect(() => {
+    data?.repository?.defaultBranchRef?.target?.oid && setOid(data?.repository?.defaultBranchRef?.target?.oid);
+  }, [data]);
+
+  return (
+    <Form
+      actions={
+        <ActionPanel>
+          <Action.SubmitForm onSubmit={handleSubmit} title="Create Branch" />
```
Comment (pernielsentikaer) [inline extensions/github/src/create-branch.tsx:90]: Could be nice to have the icon here too (icon={{ source: "branch.svg", tintColor: Color.PrimaryText }})

File: extensions/github/src/create-branch.tsx [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+        const payload = { repositoryId: values.repository, issueId: values.issue, oid };
+        const createResult = await github.createLinkedBranch({
+          input: values.name === "" ? payload : { ...payload, name: values.name },
+        });
+        const branchName = createResult.createLinkedBranch?.linkedBranch?.ref?.name;
+
+        toast.style = Toast.Style.Success;
+        toast.title = `${branchName} is created`;
+
+        if (branchName) {
```
Comment (pernielsentikaer) [inline extensions/github/src/create-branch.tsx:41]: Could we add a button to open in Browser too?

**Review by pernielsentikaer — CHANGES_REQUESTED** [review]
Hi 👋 

Thanks for your contribution 💪 

I have now tested your extension and I have some feedback ready for you:

* Add a CHANGELOG entry

I'm looking forward to testing this extension again 🔥 

Request a new review when you are ready. Feel free to contact me here or at [Slack](https://www.raycast.com/community) if you have any questions.

**Comment by khasbilegt** [comment]
@pernielsentikaer I added the changes and hope I didn't miss anything this time.

File: extensions/github/package.json [hunk 4]
```diff
@@ -59,6 +60,12 @@
       "description": "Create an issue in one of your GitHub repositories.",
       "mode": "view"
     },
+    {
+      "name": "create-issue-branch",
+      "title": "Create Issue Branch",
```
Comment (mathieudutour) [inline extensions/github/package.json:65]: is "Issue branch" something that people know? I've never seen it before. How about having just `Create Branch` and have the `Issue` dropdown be optional?

File: extensions/github/src/create-issue-branch.tsx [hunk 5]
```diff
@@ -0,0 +1,143 @@
...
+            title={repository.nameWithOwner}
+            value={repository.id}
+            icon={{ source: repository.owner.avatarUrl, mask: Image.Mask.Circle }}
+          />
+        ))}
+      </Form.Dropdown>
+
+      <Form.Separator />
+
+      <Form.Dropdown {...itemProps.issue} title="Issue" isLoading={repositoryIsLoading}>
```
Comment (mathieudutour) [inline extensions/github/src/create-issue-branch.tsx:119]: Maybe `Linked Issue` would be more explicit?

File: extensions/github/package.json (hunk 4 above)
Comment (khasbilegt) [reply extensions/github/package.json:65]: If I'm understanding you correctly, you mean to make issueId optional in branch form? Currently, when creating a branch, an issueId is required.

File: extensions/github/src/create-issue-branch.tsx (hunk 5 above)
Comment (khasbilegt) [reply extensions/github/src/create-issue-branch.tsx:119]: It's an issue field that's going to be linked once the branch is created, so changing label to Linked Issue in the form would be little confusing?

File: extensions/github/package.json (hunk 4 above)
Comment (mathieudutour) [reply extensions/github/package.json:65]: Is it? You can create a new ref: https://docs.github.com/en/rest/git/refs?apiVersion=2022-11-28#create-a-reference without an issueID. It's only to create an "issue branch" that you need one.

Comment (khasbilegt) [reply extensions/github/package.json:65]: Oh I haven't checked their REST API.

**Comment by khasbilegt** [comment]
Hi @mathieudutour, what do you think of this new changes I made?

**Pull request merged**